hi-ml-cpath/src/health_cpath/utils/deepmil_utils.py

```diff
@@ -60,11 +64,58 @@ class EncoderParams(param.Parameterized):
     encoding_chunk_size: int = param.Integer(
         default=0, doc="If > 0 performs encoding in chunks, by enconding_chunk_size tiles " "per chunk"
     )
+    ssl_checkpoint: str = param.String(
+        default="", doc="Optional run id from which to load checkpoint if using SSLEncoder"
+    )
+
+    @property
+    def ssl_checkpoint_is_url(self) -> bool:
+        return checkpoint_is_url(self.ssl_checkpoint)
+
+    @property
+    def ssl_checkpoint_is_local_file(self) -> bool:
+        return checkpoint_is_local_file(self.ssl_checkpoint)
+
+    @property
+    def ssl_checkpoint_is_aml_run_id(self) -> bool:
+        return checkpoint_is_aml_run_id(self.ssl_checkpoint)
+
+    @property
+    def is_valid_src_checkpoint(self) -> bool:
+        if self.ssl_checkpoint:
+            return self.ssl_checkpoint_is_local_file or self.ssl_checkpoint_is_url or self.ssl_checkpoint_is_aml_run_id
+        return True
```

Comment:
This is copy/pasted from src_checkpoint, can we do better?
Provide a class that does the checking of a string that describes a checkpoint, and if OK, does the download?

Reply:
Great suggestion, will make it happen!